modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/Parser.java

```diff
@@ -90,6 +94,12 @@ private CompilationUnitNode generateCompilationUnit(PackageSourceEntry sourceEnt
             BallerinaLexer lexer = new BallerinaLexer(ais);
             CommonTokenStream tokenStream = new CommonTokenStream(lexer);
             BallerinaParser parser = new BallerinaParser(tokenStream);
+
+            // If there is already defined custom error listener, set the error handler
```

Comment:
This logic is not correct. Your custom strategy is always replaced by the default one